File: ios/GhostChat/Models/AppEnums.swift
import Foundation

enum ConnectionState: String, Codable, Equatable {
    case disconnected
    case connecting
    case signaling
    case webRTC
    case connected
    case encrypted
}

enum CallState: String, Codable, Equatable {
    case idle
    case outgoingPending
    case outgoingRinging
    case incoming
    case active
    case ended
}

enum Role: String, Codable, Equatable {
    case host
    case guest
}

enum Sender: Int, Codable, Equatable {
    case me = 0
    case peer = 1
    case system = 2
}

enum MessageType: Int, Codable, Equatable {
    case text = 0
    case file = 1
    case voice = 2
    case system = 3
}

enum MessageTTL: Int, Codable, CaseIterable, Identifiable {
    case thirtySeconds = 30
    case oneMinute = 60
    case fiveMinutes = 300
    case fifteenMinutes = 900
    case oneHour = 3600

    var id: Int { rawValue }

    var localizedKey: String {
        switch self {
        case .thirtySeconds:   return "ttl.thirty_seconds"
        case .oneMinute:       return "ttl.one_minute"
        case .fiveMinutes:     return "ttl.five_minutes"
        case .fifteenMinutes:  return "ttl.fifteen_minutes"
        case .oneHour:         return "ttl.one_hour"
        }
    }
}

enum AutoLockTimeout: Int, Codable, CaseIterable, Identifiable {
    case immediate = 0
    case thirtySeconds = 30
    case oneMinute = 60
    case fiveMinutes = 300
    case thirtyMinutes = 1800

    var id: Int { rawValue }
}

File: android/app/src/main/java/com/kordar/ghostchat/models/AppEnums.kt
package com.kordar.ghostchat.models

/**
 * Mirror of iOS `Models/AppEnums.swift`. Keep cases byte-for-byte identical to
 * iOS so the two platforms can share any persisted/transport payloads.
 */

enum class ConnectionState(val wire: String) {
    DISCONNECTED("disconnected"),
    CONNECTING("connecting"),
    SIGNALING("signaling"),
    WEB_RTC("webRTC"),
    CONNECTED("connected"),
    ENCRYPTED("encrypted");

    companion object {
        fun fromWire(value: String): ConnectionState =
            values().firstOrNull { it.wire == value } ?: DISCONNECTED
    }
}

enum class CallState(val wire: String) {
    IDLE("idle"),
    OUTGOING_PENDING("outgoingPending"),
    OUTGOING_RINGING("outgoingRinging"),
    INCOMING("incoming"),
    ACTIVE("active"),
    ENDED("ended");

    companion object {
        fun fromWire(value: String): CallState =
            values().firstOrNull { it.wire == value } ?: IDLE
    }
}

enum class Role(val wire: String) {
    HOST("host"),
    GUEST("guest");

    companion object {
        fun fromWire(value: String): Role =
            values().firstOrNull { it.wire == value } ?: HOST
    }
}

enum class Sender(val raw: Int) {
    ME(0),
    PEER(1),
    SYSTEM(2);

    companion object {
        fun fromRaw(raw: Int): Sender = values().firstOrNull { it.raw == raw } ?: ME
    }
}

enum class MessageType(val raw: Int) {
    TEXT(0),
    FILE(1),
    VOICE(2),
    SYSTEM(3);

    companion object {
        fun fromRaw(raw: Int): MessageType = values().firstOrNull { it.raw == raw } ?: TEXT
    }
}

enum class MessageTTL(val seconds: Int, val localizedKey: String) {
    THIRTY_SECONDS(30, "ttl.thirty_seconds"),
    ONE_MINUTE(60, "ttl.one_minute"),
    FIVE_MINUTES(300, "ttl.five_minutes"),
    FIFTEEN_MINUTES(900, "ttl.fifteen_minutes"),
    ONE_HOUR(3600, "ttl.one_hour");

    companion object {
        fun fromSeconds(seconds: Int): MessageTTL =
            values().firstOrNull { it.seconds == seconds } ?: FIVE_MINUTES
    }
}

enum class AutoLockTimeout(val seconds: Int) {
    IMMEDIATE(0),
    THIRTY_SECONDS(30),
    ONE_MINUTE(60),
    FIVE_MINUTES(300),
    THIRTY_MINUTES(1800);

    companion object {
        fun fromSeconds(seconds: Int): AutoLockTimeout =
            values().firstOrNull { it.seconds == seconds } ?: ONE_MINUTE
    }
}
